File: 프로그래머스 고득점 kit/정렬/K번째 수.py
# 첫 시도 통과 나쁘지 않은 코드

def solution(array, commands):
    ans = []
    for i, j, k in commands:
        target = sorted(array[i - 1:j])
        ans.append(target[k - 1])

    return ans


print(solution([1, 5, 2, 6, 3, 7, 4], [
      [2, 5, 3], [4, 4, 1], [1, 7, 3]]) == [5, 6, 3])

File: 프로그래머스 고득점 kit/정렬/K번째 수.js
/** @param {number[]} array @param {number[][]} commands*/
function solution(array, commands) {
  return commands.map(([i, j, k]) => {
    const sortedArray = array.slice(i - 1, j).sort((curr, next) => curr - next)
    return sortedArray[k - 1]
  })
}
